packages/i18n-utils/src/locale-context.tsx

```diff
@@ -19,6 +20,22 @@ function getWpI18nLocaleSlug(): string | undefined {
 	return i18n.getLocaleData && i18n.getLocaleData()?.[ '' ]?.language;
 }
 
+/**
+ * Returns ISO 639 conforming locale string.
+ *
+ * @param {string} locale locale to be converted e.g. "en_US".
+ * @returns string ISO 639 locale string e.g. "en"
+ */
+function formatLocale( locale: Locale = '' ): Locale {
+	const TARGET_LOCALES = [ 'pt_br', 'pt-br', 'zh_tw', 'zh-tw', 'zh_cn', 'zh-cn' ];
```

Comment:
It's interesting that we special case only these locales. But I can see that they're the ones that we have special url support for so it's fine for our purposes. https://github.com/Automattic/wp-calypso/blob/bcadbe993db16e700d17bf910930a03040801547/packages/i18n-utils/src/locales.ts

Maybe we should add `zh_sg` to the list since it's also mentioned explicitly? https://github.com/Automattic/wp-calypso/blob/bcadbe993db16e700d17bf910930a03040801547/packages/i18n-utils/src/locales.ts#L20